pkg/operation/botanist/controlplane.go

```diff
@@ -431,12 +431,31 @@ func (b *Botanist) DeployKubeAPIServer(ctx context.Context) error {
 		hvpaEnabled               = gardenletfeatures.FeatureGate.Enabled(features.HVPA)
 		mountHostCADirectories    = gardenletfeatures.FeatureGate.Enabled(features.MountHostCADirectories)
 		memoryMetricForHpaEnabled = false
+
+		scaleDownUpdateMode       = hvpav1alpha1.UpdateModeAuto
+		minReplicas         int32 = 1
+		maxReplicas         int32 = 4
 	)
 
 	if b.ManagedSeed != nil {
 		// Override for shooted seeds
 		hvpaEnabled = gardenletfeatures.FeatureGate.Enabled(features.HVPAForShootedSeed)
 		memoryMetricForHpaEnabled = true
+
+		if b.ManagedSeedAPIServer != nil {
+			autoscaler := b.ManagedSeedAPIServer.Autoscaler
+			minReplicas = *autoscaler.MinReplicas
+			maxReplicas = autoscaler.MaxReplicas
+		}
+	}
+
+	if b.Shoot.Purpose == gardencorev1beta1.ShootPurposeProduction {
+		minReplicas = 2
```

Comment:
Though less likely, do we need to make sure that `minReplicas <= maxReplicas` here? There is a small chance that the [managed seed case above](https://github.com/gardener/gardener/commit/8084137055199030c7c2d6fe153070ca39078ed8#diff-201c3f0c956474991f7e0ebb09d5cb2c23b1b8dbfdad19c36c81c188ce1f13a5R445-R449) might have configured maxReplicas different from the default value of `4`.

Reply:
I've swapped the statements so that the min > max cannot happen anymore.